### PR TITLE
Feature/212

### LINKED ISSUE #212: Implement ranking strategies as followed by ConvKB and by Old Dogs New tricks paper
**Background and Context**
In AmpliGraph, while evaluating corruptions, if the test set triple gets same score as any of the corruptions, we assign the worst rank. There are other approaches followed in literature. We should implement all these strategies to allow the user to compare model performance using all three approaches.
**Description**
Let's look at each of the three strategies in detail with an example. 

Assume there are only 10 corruptions, and assume that all the corruptions get the same score as the test triple. The ranks assigned by the three strategies are as follows: 
- Assign the **worst rank** i.e. the test set triple gets a rank of 11. This is followed by most papers in the literature. This is the strictest approach and it drives down the mrr by a large margin if there are many ties.
- Assign the **middle rank** i.e. the test set triple gets a rank of 6. We found this strategy being used by ICLR 2020 paper<sup>1</sup>. This approach seems to be fair towards the model in resolving the ties as it assigns the middle rank to break ties.
- Assign the **best rank** i.e. the test set triple gets a rank of 1. This approach is followed by ConvKB<sup>2</sup>.  This approach is overly biased and helps the model achieve a very good mrr in case of ties.
 
References:
1. Ruffinelli et al., You CAN Teach an Old Dog New Tricks! On Training Knowledge Graph Embeddings, ICLR 2020
2. Nguyen et al., A Novel Embedding Model for Knowledge Base Completion Based on Convolutional Neural Network, NAACL 2018

### PR DESCRIPTION
**Pull request by sumitpai**
#### Related Issue(s)

Fixes #212 
#### Description of Changes
Added a function to compare scores based on specified strategy.

### CODE REVIEW

File: ampligraph/evaluation/protocol.py [hunk 1]
```diff
@@ -444,7 +444,7 @@ def to_idx(X, ent_to_idx, rel_to_idx):
 
 
 def evaluate_performance(X, model, filter_triples=None, verbose=False, filter_unseen=True, entities_subset=None,
-                         corrupt_side='s,o', use_default_protocol=False):
+                         corrupt_side='s,o', ranking_strategy='worst', use_default_protocol=False):
```
Comment (NicholasMcCarthy) [inline ampligraph/evaluation/protocol.py:447]: A side comment, but I thought we had removed the use_default_protocol parameter at some point?

Comment (sumitpai) [reply ampligraph/evaluation/protocol.py:447]: We haven't yet.

**Review by NicholasMcCarthy — APPROVED** [review]
